code/modules/detectivework/tools/rag.dm

```diff
@@ -156,10 +156,19 @@
 		new /obj/effect/decal/cleanable/ash(get_turf(src))
 		qdel(src)
 
-//rag must have a minimum of 2 units welder fuel and at least 80% of the reagents must be welder fuel.
-//maybe generalize flammable reagents someday
+//rag must have a minimum of 2 units welder fuel or ehtanol based reagents and at least 80% of the reagents must so.
 /obj/item/weapon/reagent_containers/glass/rag/proc/can_ignite()
-	var/fuel = reagents.get_reagent_amount("fuel")
+	var/fuel
+	if(reagents.get_reagent_amount("fuel"))
+		fuel += reagents.get_reagent_amount("fuel")
+		world << fuel
+
+	else
+		for(var/datum/reagent/ethanol/R in reagents.reagent_list)
+			if(istype(R, /datum/reagent/ethanol))
```

Comment:
This `istype()` is extraneous. If you specify a type using `for(type in list)`, you'll only get said type.